packages/rs-drive-abci/src/abci/app/consensus.rs

```diff
@@ -17,16 +19,19 @@ use tenderdash_abci::proto::abci as proto;
 pub struct ConsensusAbciApplication<'a, C> {
     /// Platform
     platform: &'a Platform<C>,
-    /// The current transaction
-    transaction: RwLock<Option<Transaction<'a>>>,
+    /// The current GroveDb transaction
+    transaction: RefCell<Option<Transaction<'a>>>,
```

Comment:
We should not change this to a RefCell. Doing so would make it so we could never use async.

Reply:
Reverted to RwLock. Unfortunately, RwLock doesn't help. To be able to pass it between threads it must be Mutex :(. I hope there are other options what I don't know yet.